src/main/java/org/jabref/gui/preferences/groups/GroupsTabViewModel.java

```diff
@@ -22,23 +22,20 @@ public GroupsTabViewModel(GroupsPreferences groupsPreferences) {
 
     @Override
     public void setValues() {
-        switch (groupsPreferences.getGroupViewMode()) {
-            case INTERSECTION -> {
-                groupViewModeIntersectionProperty.setValue(true);
-                groupViewModeUnionProperty.setValue(false);
-            }
-            case UNION -> {
```

Comment:
Keep the switch statement. By "else" a future contributor will otherwise never know by compile error, if he missed something. Use explicit cases, whenever possible.
Use `case FILTER, INVERT ->` here.

Reply:
I only need to know if the mode is INTERSECTION or UNION. The `getGroupViewMode` method returns a set, and I want to check if INTERSECTION exists. The else handles the UNION case.
Removed the UNION flag because it’s the negation of INTERSECTION. By knowing one, you can infer the other. Instead of checking for both INTERSECTION and UNION each time, and set one to true and the other to false.

Reply:
And someday another option is added by antoher programmer we are not thinking of yet. So either case `case FILTER, INVERT ->` (`default` would be the same as else). Use explicit cases, so that in the future, if someone is adding a fourth option to GroupViewMode, the compiler will make him decide on this switch here.

Reply:
If someone adds a new flag and wants to handle it here, they will use another "if" branch. Similarly, if I want to do something with FILTER, I will do it like this:

```java
// INTERSECTION or UNION
if (groupsPreferences.getGroupViewMode().contains(GroupViewMode.INTERSECTION)) {
    groupViewModeIntersectionProperty.setValue(true);
    groupViewModeUnionProperty.setValue(false);
} else {
    groupViewModeIntersectionProperty.setValue(false);
    groupViewModeUnionProperty.setValue(true);
}
// INVERT
if (groupsPreferences.getGroupViewMode().contains(GroupViewMode.INVERT)) {
    // do action
}
// FILTER
if (groupsPreferences.getGroupViewMode().contains(GroupViewMode.FILTER)) {
    // do action
}
```
This is because the group view mode can have **both** FILTER and INVERT at the same time.

Reply:
ok, I see. I thought the action taken was mutually exclusive.